opencog/openpsi/README.md

```diff
@@ -142,17 +175,35 @@ Coming soon :smile:
         (Present value = 0 or 1)
     STIcga = short-term-importance of the psi-rule (Present value = 1)
 
-XXX FIXME -- this needs fixing, as contexts may contain variables whose
-groundings can carry over into the action.  In addition, it is extremely
-inefficient if there are many rules; thus the DualLink can be (should
-be) used to narrow down the search for contextts.  A crude prototype of
-this has been implemented in the nlp/aiml subsysgtem, and needs to be
-generalized and ported over here.
-
-4. action execution and goal evaluation:
-  * After the rules are selected, then the action and goals are extracted.
-    During the goal extraction, every goal that is related through the
-    action are choosen.
+4. Action Selection bugs. XXX FIXME -- The current implementation has
+   multiple implementation issues. These include:
+
+ * If a context contains variables that also appear in the action,
+   then these are not handled correctly. For example, if the context
+   is "variable $X is flying towards me", and the action is "catch
+   variable $X", the current satisfiability code can determine if
+   the context holds true, by grounding variable $X, but that
+   grounding is not passed to the action. That is, when the action is
+   executed, teh variable $X is not known/grounded.
+
+ * The default action selector is extremely inefficient if there
+   are more than a few hundred rules, as it attempts to find and
+   work with all of them. This overhead can be avoided by using the
+   DualLink for narrowing the search for applicable contexts.  There
+   is a protottype showing the DualLink in action, in the nlp/aiml
+   subsystem. (There are more than 100K AIML rules; the DualLink
+   narrows these down to the handful of rules applicable to the
+   current context). This code needs to be prted here.
```

Comment:
Yes the dual link will work for contexts that are used in aiml's psi-rules, but wouldn't work for the rules in ros-behavior-scripting.